### DISCUSSION TITLE
Installing batchrunner.py

### DISCUSSION THREAD

**Discussion by AlABarazi** [mesa/mesa#1415] [Q&A] [answered]
Hi,
I am trying to install the last version of mesa with the new batch_run  function  (BatchRunner class depreciated) but I am not able to get the last version of batchrunner file. I am using this command $ pip install -e git+https://github.com/projectmesa/mesa#egg=mesa

**Comment by rht** [comment]
You can install the latest release to get `batch_run`. But if you still want to install the latest development branch, it is likely that your local Git cache it outdated. You might have to do `pip install -U -e git+https://github.com/projectmesa/mesa#egg=mesa`. `-U` for force update.

**Reply by AlABarazi** [reply]
Thanks for the suggestion but it didn't work . I am getting ImportError: cannot import name 'batch_run'

`
(base) ➜  ~ source activate /Users/alaabarazi/anaconda3/envs/ml               
(ml) ➜  ~ pip install -U -e git+https://github.com/projectmesa/mesa#egg=mesa
Obtaining mesa from git+https://github.com/projectmesa/mesa#egg=mesa
  Updating ./src/mesa clone
  Running command git fetch -q --tags
  Running command git reset --hard -q bb56278d6dfbc20ecd38b5b9cfe3c3ca8ff7811a
  Preparing metadata (setup.py) ... done
Requirement already satisfied: click in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (7.0)
Requirement already satisfied: cookiecutter in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (1.7.0)
Requirement already satisfied: networkx in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (2.4)
Requirement already satisfied: numpy in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (1.15.4)
Requirement already satisfied: pandas in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (0.24.2)
Requirement already satisfied: tornado in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (5.1.1)
Requirement already satisfied: tqdm in ./anaconda3/envs/ml/lib/python3.6/site-packages (from mesa) (4.45.0)
Requirement already satisfied: jinja2-time>=0.1.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (0.2.0)
Requirement already satisfied: requests>=2.18.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (2.21.0)
Requirement already satisfied: whichcraft>=0.4.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (0.6.1)
Requirement already satisfied: binaryornot>=0.2.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (0.4.4)
Requirement already satisfied: jinja2>=2.7 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (2.10)
Requirement already satisfied: future>=0.15.2 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (0.17.1)
Requirement already satisfied: poyo>=0.1.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from cookiecutter->mesa) (0.5.0)
Requirement already satisfied: decorator>=4.3.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from networkx->mesa) (4.3.0)
Requirement already satisfied: pytz>=2011k in ./anaconda3/envs/ml/lib/python3.6/site-packages (from pandas->mesa) (2018.9)
Requirement already satisfied: python-dateutil>=2.5.0 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from pandas->mesa) (2.7.5)
Requirement already satisfied: chardet>=3.0.2 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from binaryornot>=0.2.0->cookiecutter->mesa) (3.0.4)
Requirement already satisfied: MarkupSafe>=0.23 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from jinja2>=2.7->cookiecutter->mesa) (1.1.0)
Requirement already satisfied: arrow in ./anaconda3/envs/ml/lib/python3.6/site-packages (from jinja2-time>=0.1.0->cookiecutter->mesa) (0.15.5)
Requirement already satisfied: six>=1.5 in ./.local/lib/python3.6/site-packages (from python-dateutil>=2.5.0->pandas->mesa) (1.14.0)
Requirement already satisfied: certifi>=2017.4.17 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from requests>=2.18.0->cookiecutter->mesa) (2020.6.20)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from requests>=2.18.0->cookiecutter->mesa) (1.24.1)
Requirement already satisfied: idna<2.9,>=2.5 in ./anaconda3/envs/ml/lib/python3.6/site-packages (from requests>=2.18.0->cookiecutter->mesa) (2.8)
Installing collected packages: mesa
  Attempting uninstall: mesa
    Found existing installation: Mesa 0.8.6
    Uninstalling Mesa-0.8.6:
      Successfully uninstalled Mesa-0.8.6
  Running setup.py develop for mesa
Successfully installed mesa-0.8.6
(ml) ➜  ~ python                                                            
Python 3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 14:01:38) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from mesa.batchrunner import batch_run
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'batch_run'
>>> 
`

**Reply by rht** [reply]
0.8.6 is old. Current latest version is 1.0.0.

**Comment by rht** [comment, marked as answer]
Try `pip install -U -e git+https://github.com/projectmesa/mesa@main#egg=mesa`.

**Reply by AlABarazi** [reply]
Thanks for your help. The problem was that I have Python 3.6 and mesa want >=3.7

**Reply by rht** [reply]
OK glad you fixed it. But the 3.6 issue was uncovered because this `pip install` finally installed from the latest main, and it raised this separate new error? There were 2 problems, then. Does this particular answer count as the solution for the first problem?

**Comment by AlABarazi** [comment]
Yes exactly. Yes it is solved thanks 🙏🏻
